src/datumaro/experimental/dataset.py

```diff
@@ -469,6 +471,34 @@ def convert_to_schema(
             categories=inferred_categories,
         )
 
+    def filter_by_subset(self, subset: Subset) -> Dataset[DType]:
+        """
+        Return new dataset with items from given subset.
+
+        Args:
+            subset: the subset to filter on
+
+        Returns:
+            A new Dataset with items of the given subset.
+        """
+        subset_column_name: str | None = None
+        for column_name, attribute_info in self.schema.attributes.items():
+            if isinstance(attribute_info.annotation, SubsetField):
+                subset_column_name = column_name
+                break
+        if subset_column_name is None:
+            raise RuntimeError(
```

Comment:
```suggestion
        for column_name, attribute_info in self.schema.attributes.items():
            if isinstance(attribute_info.annotation, SubsetField):
                subset_column_name = column_name
                break
        else:
            raise RuntimeError(
```
I know that not everyone like `for...else`, feel free to ignore.